service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/AmqpConnectionHandler.java

```diff
@@ -148,7 +204,15 @@ public void onSessionRemoteOpen(Event event)
     public void onConnectionRemoteOpen(Event event)
     {
         super.onConnectionRemoteOpen(event);
+        this.connection = event.getConnection();
         this.connectionOpenedRemotely = true;
+
+        Session cbsSession = event.getConnection().session();
```

Comment:
on remote connection open, will the cbs session be the only session that is returned by protonj?

Reply:
The call to connection.session(); actually creates a session, so we get to make it into the CBS session after it is created

Reply:
It can be read as "for the connection that this event is tied to, create a session."

Reply:
oh wow, I read `event.getConnection().session()` as a getter! 
might be useful to add this as a comment here